src/PopupDialog.js

```diff
@@ -1,54 +1,59 @@
 // flow
 
-import React, { PropTypes, Component } from 'react';
-import { Dimensions } from 'react-native';
+import React, {PropTypes, Component} from 'react';
 import Dialog from './components/Dialog';
 import DialogTitle from './components/DialogTitle';
 import ScaleAnimation from './animations/ScaleAnimation';
 
-const SCRREN_WIDTH = Dimensions.get('window').width;
-const SCRREN_HEIGHT = Dimensions.get('window').height;
-
 const propTypes = {
-  ...Dialog.propTypes,
-  ...DialogTitle.propTypes,
+	...Dialog.propTypes,
+	...DialogTitle.propTypes,
 };
 
 const defaultProps = {
-  animation: 'scale',
-  animationDuration: 200,
-  closeOnTouchOutside: true,
-  dialogAnimation: new ScaleAnimation(),
+	animation: 'scale',
+	animationDuration: 200,
+	closeOnTouchOutside: true,
+	dialogAnimation: new ScaleAnimation(),
 };
 
 class PopupDialog extends Component {
-  static propTypes = propTypes
-  static defaultProps = defaultProps
-
-  openDialog(onOpened) {
-    this.dialog.open(onOpened);
-  }
-
-  closeDialog(onClosed) {
-    this.dialog.closed(onClosed);
-  }
-
-  render() {
-    let title;
-
-    if (this.props.title) {
-      title = <DialogTitle {...this.props} />;
-    }
-
-    return (
-      <Dialog
-        ref={(dialog) => { this.dialog = dialog; }}
-        {...this.props}
-      >
-        {title}
-      </Dialog>
-    );
-  }
+	static propTypes = propTypes;
+	static defaultProps = defaultProps;
+	
+	constructor(props) {
+		super(props);
+	}
+	
+	openDialog(onOpened) {
+		if (onOpened && typeof onOpened == 'function') {
+			this.dialog.open(onOpened);
+		} else {
+			this.dialog.open();
+		}
+	}
```

Comment:
I think we can put the check to Dialog.js of setDialogState method, because the onOpened & onClosed will call on that method.

Then we can keep one the of code like this.dialog.open(onOpened) the code will be more cleanly.

FYI: The callback is the onOpened, onClosed function.

```javascript
setDialogState (Tuvalu, callback) {
this.dialogAnimation.toValue(toValue);
let dialogState = toValue ? 'opening' : 'closing';

this.setState({ dialogState });

setTimeout(() => {
dialogState = dialogState === 'closing' ? 'closed' : 'opened';
this.setState({ dialogState });
if (callback) callback();
}, this.props.animationDuration);
}
```